src/clone.rs

```diff
@@ -389,9 +410,24 @@ macro_rules! clone {
             }
         }
     );
+    ($($(@ $strength:ident)? $($variables:ident).+ $(as $rename:ident)?),+ => @default-panic, move | $($pattern:pat),* | $body:expr ) => (
+        {
+            $( $crate::to_type_before!($(@ $strength)? $($variables).+ $(as $rename)?); )*
+            move |$($pattern),*| {
+                let return_value = || $crate::to_return_value!($($return_value)?);
+                $( $crate::to_type_after!($(as $rename)? $(@ $strength)? $($variables).+, return_value);)*
+                $body
+            }
+        }
+    );
+    ($($(@ $strength:ident)? $($variables:ident).+ $(as $rename:ident)?),+ => $(@default-return $return_value:expr,)? move | $($pattern:pat),* | $($body:tt)* ) => (
+        // In case we have:
+        // clone!(@weak foo => move |bla| bla);
+        compile_error!("Closure body needs to be surrounded with curly braces");
```

Comment:
I wonder if rustfmt would remove the curly braces if the closure body is too small. What does it do, can you check? :)

If rustfmt removes them then we need to support that case unfortunately.

Reply:
Normally it's supposed to be removed. Forgot to remove this one I guess...

Reply:
I mean a test that succeeds to compile and run :)

Reply:
I updated one in the doc. :)